### PR TITLE
Builtin/../falling: Fix bugs caused by 'ignore' nodes

### PR DESCRIPTION
**Pull request by paramat**
Original commit by t4im, rebased and developed by paramat.
Fix CONTENT_IGNORE being replaced by falling nodes or causing large
areas of sand to collapse into itself.
Format some conditional code for clarity.
Add and clarify some comments.
//////////////////////////////////////////////////////

New PR for #4630 as t4im is absent, needed before we can make 'nodeupdate' official API (which i will do next).
See original PR for discussion and more details. Original was confirmed as fixing #4594 
Addresses #3398 and #4594 

In 2 cases this PR adds 'get node or nil' to prevent problem behaviours happening when ignore is below.
In 'on step', falling nodes will no longer remove ignore below, this happened because ignore is 'buildable to'.
Falling nodes will no longer start to fall if ignore is below, causing sand collapsing into itself at mapblock borders.

### CODE REVIEW

File: builtin/game/falling.lua [hunk 1]
```diff
@@ -154,16 +155,16 @@ function nodeupdate_single(p)
...
 		local p_bottom = {x = p.x, y = p.y - 1, z = p.z}
-		local n_bottom = core.get_node(p_bottom)
-		local d_bottom = core.registered_nodes[n_bottom.name]
+		local n_bottom = core.get_node_or_nil(p_bottom)
+		local d_bottom = n_bottom and core.registered_nodes[n_bottom.name]
 		-- Note: walkable is in the node definition, not in item groups
-		if d_bottom and
+		if n_bottom and d_bottom and
 				(core.get_item_group(n.name, "float") == 0 or
-					d_bottom.liquidtype == "none") and
```
Comment (SmallJoker) [inline builtin/game/falling.lua]: Everything on the same level of indentation makes it hard to read the logic.

Comment (paramat) [inline builtin/game/falling.lua]: The previous indentation was worse.

Comment (paramat) [inline builtin/game/falling.lua]: I oculd perhaps reformat and add some empty lines for clarity.

Comment (paramat) [inline builtin/game/falling.lua]: Ok.

**Comment by paramat** [comment]
Reformatted conditional for clarity.
Looks a little weird but it's strictly split into the primary 'and' blocks.

**Comment by paramat** [comment]
Tested. Falling behaviour seems ok and the sand collapse bug seems fixed.

**Comment by paramat** [comment]
```
		local n_bottom = core.get_node_or_nil(p_bottom)
		local d_bottom = n_bottom and core.registered_nodes[n_bottom.name]
		-- Note: walkable is in the node definition, not in item groups
		if n_bottom and

				d_bottom and
```
I think we don't need to check both 'n_bottom' and 'd_bottom', if 'n_bottom' is 'nil' for an unloaded area then 'd_bottom' will be 'false'. If 'n_bottom' is not 'nil' then 'd_bottom' will always return a nodedef because all nodes, even 'unknown' have nodedefs.

**Comment by paramat** [comment]
Updated and improved.

**Comment by paramat** [comment]
3c5da70d7cfafe52f286bbe9a36ea7c04e92bf9f

**Pull request closed without merging**